lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -296,10 +303,33 @@ def commit(request)
           response,
           authorization: authorization_from(response),
           test: test?,
-          error_code: error_code_from(response)
+          error_code: error_code_from(response),
+          response_type: response_type(response[:response_code]&.to_i),
```

Comment:
Are you sure, we have response[:response_code] for the code in the response hash?

Reply:
not, yet, bacause I can't test the API
it can be replaced with dig("result", "errors", "error", "code") in the future